### PR TITLE
Appstudio tier

### PR DESCRIPTION
**Pull request by alexeykazakov**
Tests for codeready-toolchain/host-operator#534

### CODE REVIEW

File: testsupport/tiers/checks.go [hunk 1]
```diff
@@ -271,6 +275,57 @@ func (a *testTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
 	return []clusterObjectsCheck{}
 }
 
+type appstudioTierChecks struct {
+	baseTierChecks
+}
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:280]: The `appstudio` tier is not based on the `base` tier, so we should not reuse it
```suggestion
type appstudioTierChecks struct {
	tierName string
}
```

File: testsupport/tiers/checks.go [hunk 2]
```diff
@@ -68,6 +69,9 @@ func NewChecks(tier string) (TierChecks, error) {
 	case advanced:
 		return &advancedTierChecks{baseTierChecks{tierName: advanced}}, nil
 
+	case appstudio:
+		return &appstudioTierChecks{baseTierChecks{tierName: appstudio}}, nil
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:73]:
```suggestion
		return &appstudioTierChecks{tierName: appstudio}, nil
```

File: testsupport/tiers/checks.go [hunk 3]
```diff
@@ -327,6 +382,20 @@ func rbacEditRoleBinding() namespaceObjectsCheck {
...
 }
 
+func userViewRoleBinding() namespaceObjectsCheck {
+	return func(t *testing.T, ns *v1.Namespace, memberAwait *wait.MemberAwaitility, userName string) {
+		rb, err := memberAwait.WaitForRoleBinding(ns, "user-view")
+		require.NoError(t, err)
+		assert.Len(t, rb.Subjects, 1)
+		assert.Equal(t, "User", rb.Subjects[0].Kind)
+		assert.Equal(t, userName, rb.Subjects[0].Name)
+		assert.Equal(t, "edit", rb.RoleRef.Name)
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:392]: This is supposed to be `view` and not `edit`, right?
```suggestion
		assert.Equal(t, "view", rb.RoleRef.Name)
```

File: testsupport/tiers/checks.go [hunk 4]
```diff
@@ -271,6 +275,57 @@ func (a *testTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
...
+type appstudioTierChecks struct {
+	baseTierChecks
+}
+
+func (a *appstudioTierChecks) GetTierObjectChecks() []tierObjectCheck {
+	return []tierObjectCheck{nsTemplateTier(a.tierName, 30)}
+}
+
+func (a *appstudioTierChecks) GetNamespaceObjectChecks(nsType string) []namespaceObjectsCheck {
+	checks := append(commonChecks,
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:287]: you cannot use the `commonChecks` as they expect `user-edit` RoleBinding which is not part of this tier

File: testsupport/tiers/checks.go [hunk 5]
```diff
@@ -271,6 +275,57 @@ func (a *testTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
...
+		userSaReadRoleBinding(),
+		execPodsRole(),
+		toolchainSaReadRole(),
+		appstudioServiceAccount(),
+		numberOfToolchainRoles(2),
+		numberOfToolchainRoleBindings(5),
+		numberOfToolchainServiceAccounts(1),
+	)
+
+	checks = append(checks, commonNetworkPolicyChecks()...)
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:302]: `commonNetworkPolicyChecks` doesn't contain the networkPolicy for CRW. I should also check the number of present networkPolicies
```suggestion
	checks = append(checks, append(commonNetworkPolicyChecks(), networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))...)
```

**Review by xcoulon — APPROVED** [review]

**Pull request merged**